Canot use Kali anymore

I usually use Kali.
I compiled Speech-Dispatcher master (the version I have is 0.10.1-178-gdb63) and only Voxin can be used with Orca.
I have make de configuration in .config/speech-dispatcher I removed to test.
After removing them, if I select Kali on Orca settings, Speech-dispatcher falls back to Espeak.

The logs with my config files:
* [kali.log](https://github.com/brailcom/speechd/files/6171617/kali.log)
* [speech-dispatcher.log](https://github.com/brailcom/speechd/files/6171618/speech-dispatcher.log)
* [voxin.log](https://github.com/brailcom/speechd/files/6171619/voxin.log)

Without my custom files:
* [baratinoo.log](https://github.com/brailcom/speechd/files/6171635/baratinoo.log)
* [espeak.log](https://github.com/brailcom/speechd/files/6171636/espeak.log)
* [espeak-ng.log](https://github.com/brailcom/speechd/files/6171639/espeak-ng.log)
* [espeak-mbrola-generic.log](https://github.com/brailcom/speechd/files/6171637/espeak-mbrola-generic.log)
* [flite.log](https://github.com/brailcom/speechd/files/6171640/flite.log)
* [kali.log](https://github.com/brailcom/speechd/files/6171641/kali.log)
* [pico.log](https://github.com/brailcom/speechd/files/6171643/pico.log)
* [speech-dispatcher.log](https://github.com/brailcom/speechd/files/6171644/speech-dispatcher.log)
* [voxin.log](https://github.com/brailcom/speechd/files/6171645/voxin.log)

I ran Speech-Dispatcher in debug mode to make these logs.

The things I configured are:
* Deactivate abbreviations in Kali and Voxin&IBMTTS.
* Set symbolsPreproc to "all".

I tried to clear all my previous installation before re-installing Speech-Dispatcher with no success.

Yes, it worked on 0.10.0.

While looking for the revision which this bug started from, I encountered a compilation error from a82eb3d71fe54aea35a78a1f740345e870bb517b
```
  CXX      sd_kali-kali.o
In file included from module_utils.h:44,
                 from kali.cpp:38:
module_main.h:140:6: error: conflicting declaration of ‘void module_report_index_mark(char*)’ with ‘C’ linkage
 void module_report_index_mark(char *mark);
      ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from module_utils.h:42,
                 from kali.cpp:38:
../../src/common/common.h:32:6: note: previous declaration with ‘C++’ linkage
 void module_report_index_mark(char *mark);
      ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from module_utils.h:44,
                 from kali.cpp:38:
module_main.h:142:6: error: conflicting declaration of ‘void module_report_event_begin()’ with ‘C’ linkage
 void module_report_event_begin(void);
      ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from module_utils.h:42,
                 from kali.cpp:38:
../../src/common/common.h:34:6: note: previous declaration with ‘C++’ linkage
 void module_report_event_begin(void);
      ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from module_utils.h:44,
                 from kali.cpp:38:
module_main.h:144:6: error: conflicting declaration of ‘void module_report_event_end()’ with ‘C’ linkage
 void module_report_event_end(void);
      ^~~~~~~~~~~~~~~~~~~~~~~
In file included from module_utils.h:42,
                 from kali.cpp:38:
../../src/common/common.h:36:6: note: previous declaration with ‘C++’ linkage
 void module_report_event_end(void);
      ^~~~~~~~~~~~~~~~~~~~~~~
In file included from module_utils.h:44,
                 from kali.cpp:38:
module_main.h:146:6: error: conflicting declaration of ‘void module_report_event_stop()’ with ‘C’ linkage
 void module_report_event_stop(void);
      ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from module_utils.h:42,
                 from kali.cpp:38:
../../src/common/common.h:38:6: note: previous declaration with ‘C++’ linkage
 void module_report_event_stop(void);
      ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from module_utils.h:44,
                 from kali.cpp:38:
module_main.h:148:6: error: conflicting declaration of ‘void module_report_event_pause()’ with ‘C’ linkage
 void module_report_event_pause(void);
      ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from module_utils.h:42,
                 from kali.cpp:38:
../../src/common/common.h:40:6: note: previous declaration with ‘C++’ linkage
 void module_report_event_pause(void);
      ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from module_utils.h:44,
                 from kali.cpp:38:
module_main.h:150:6: error: conflicting declaration of ‘void module_report_icon(const char*)’ with ‘C’ linkage
 void module_report_icon(const char *icon);
      ^~~~~~~~~~~~~~~~~~
In file included from module_utils.h:42,
                 from kali.cpp:38:
../../src/common/common.h:42:6: note: previous declaration with ‘C++’ linkage
 void module_report_icon(const char *icon);
      ^~~~~~~~~~~~~~~~~~
In file included from kali.cpp:38:
module_utils.h:159:6: error: conflicting declaration of ‘void set_speaking_thread_parameters()’ with ‘C’ linkage
 void set_speaking_thread_parameters();
      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from module_utils.h:42,
                 from kali.cpp:38:
../../src/common/common.h:29:6: note: previous declaration with ‘C++’ linkage
 void set_speaking_thread_parameters(void);
      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
kali.cpp: In function ‘int module_init(char**)’:
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:109:2: note: in expansion of macro ‘DBG’
  DBG("Module init");
  ^~~
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:128:2: note: in expansion of macro ‘DBG’
  DBG("KaliMaxChunkLength = %d\n", KaliMaxChunkLength);
  ^~~
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:129:2: note: in expansion of macro ‘DBG’
  DBG("KaliDelimiters = %s\n", KaliDelimiters);
  ^~~
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:130:2: note: in expansion of macro ‘DBG’
  DBG("KaliExpandAbbreviations = %d\n", KaliExpandAbbreviations);
  ^~~
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:136:2: note: in expansion of macro ‘DBG’
  DBG("Kali: creating new thread for kali_speak\n");
  ^~~
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:140:3: note: in expansion of macro ‘DBG’
   DBG("Kali: thread failed\n");
   ^~~
kali.cpp: In function ‘int module_speak(gchar*, size_t, SPDMessageType)’:
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:163:2: note: in expansion of macro ‘DBG’
  DBG("write()\n");
  ^~~
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:166:3: note: in expansion of macro ‘DBG’
   DBG("Speaking when requested to write");
   ^~~
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:170:2: note: in expansion of macro ‘DBG’
  DBG("Requested data: |%s|\n", data);
  ^~~
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:191:2: note: in expansion of macro ‘DBG’
  DBG("Kali: leaving write() normally\n\r");
  ^~~
kali.cpp: In function ‘int module_stop()’:
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:198:2: note: in expansion of macro ‘DBG’
  DBG("kali: stop()\n");
  ^~~
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:202:3: note: in expansion of macro ‘DBG’
   DBG("Stopping audio");
   ^~~
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:205:4: note: in expansion of macro ‘DBG’
    DBG("WARNING: Non 0 value from spd_audio_stop: %d",
    ^~~
kali.cpp: In function ‘size_t module_pause()’:
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:214:2: note: in expansion of macro ‘DBG’
  DBG("pause requested\n");
  ^~~
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:216:3: note: in expansion of macro ‘DBG’
   DBG("Kali doesn't support pause, stopping\n");
   ^~~
kali.cpp: In function ‘int module_close()’:
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:229:2: note: in expansion of macro ‘DBG’
  DBG("kali: close()\n");
  ^~~
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:231:2: note: in expansion of macro ‘DBG’
  DBG("Stopping speech");
  ^~~
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:241:2: note: in expansion of macro ‘DBG’
  DBG("Terminating threads");
  ^~~
kali.cpp: In function ‘void* _kali_speak(void*)’:
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:266:2: note: in expansion of macro ‘DBG’
  DBG("kali: speaking thread starting.......\n");
  ^~~
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:272:3: note: in expansion of macro ‘DBG’
   DBG("Semaphore on\n");
   ^~~
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:283:5: note: in expansion of macro ‘DBG’
     DBG("Stop in child, terminating");
     ^~~
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:294:5: note: in expansion of macro ‘DBG’
     DBG("End of message");
     ^~~
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:301:4: note: in expansion of macro ‘DBG’
    DBG("Returned %d bytes from get_part\n", bytes);
    ^~~
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:302:4: note: in expansion of macro ‘DBG’
    DBG("Text to synthesize is '%s'\n", buf);
    ^~~
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:305:5: note: in expansion of macro ‘DBG’
     DBG("Pause requested in parent, position %d\n",
     ^~~
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:313:5: note: in expansion of macro ‘DBG’
     DBG("Speaking in child...");
     ^~~
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:315:5: note: in expansion of macro ‘DBG’
     DBG("Trying to synthesize text");
     ^~~
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:324:6: note: in expansion of macro ‘DBG’
      DBG("Stop in child, terminating");
      ^~~
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:336:5: note: in expansion of macro ‘DBG’
     DBG("Got %d samples", track.num_samples);
     ^~~
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:339:7: note: in expansion of macro ‘DBG’
       DBG("Stop in child, terminating");
       ^~~
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:344:6: note: in expansion of macro ‘DBG’
      DBG("Playing part of the message");
      ^~~
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:347:7: note: in expansion of macro ‘DBG’
       DBG("ERROR: failed to play the track");
       ^~~
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:349:7: note: in expansion of macro ‘DBG’
       DBG("Stop in child, terminating (s)");
       ^~~
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:356:5: note: in expansion of macro ‘DBG’
     DBG("End of data in speaking thread");
     ^~~
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:367:5: note: in expansion of macro ‘DBG’
     DBG("Stop in child, terminating");
     ^~~
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:379:2: note: in expansion of macro ‘DBG’
  DBG("kali: speaking thread ended.......\n");
  ^~~
kali.cpp: In function ‘SPDVoice** kali_get_voices()’:
../../src/common/common.h:27:31: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
 #define DBG(arg...) MSG(4, arg)
                               ^
kali.cpp:472:2: note: in expansion of macro ‘DBG’
  DBG("Kali: %d voices total.", num_voices);
  ^~~
make[3]: *** [Makefile:1464: sd_kali-kali.o] Error 1
make[3] : on quitte le répertoire « /home/pzajda/git/speechd/src/modules »
make[2]: *** [Makefile:453: all-recursive] Error 1
make[2] : on quitte le répertoire « /home/pzajda/git/speechd/src »
make[1]: *** [Makefile:587: all-recursive] Error 1
make[1] : on quitte le répertoire « /home/pzajda/git/speechd »
make: *** [Makefile:475: all] Error 2
```

For this I made a `git reset --hard && git checkout <revision>` and to compile I called `make clean && make maintainer-clean && ./build.sh && ./configure && make && sudo make install` to be sure to have the most clean Git repository.
When I opened this issue I did not call all these commands, I began with `make` with no compilation issue.
Which could explain why this bug?
Do you think it could be useful to continue trying to compile next revisions to see if something changes?

--- a/src/modules/kali.cpp
+++ b/src/modules/kali.cpp
@@ -158,7 +158,7 @@ SPDVoice **module_list_voices(void)
 	return kali_voice_list;
 }
 
-int module_speak(gchar * data, size_t bytes, SPDMessageType msgtype)
+int module_speak(const char * data, size_t bytes, SPDMessageType msgtype)
 {
 	DBG("write()\n");
 

